### PR TITLE
Core Concepts Of Redux Store Management

### LINKED ISSUE #3486: Core Concepts Of Redux Store Management
Redux happens to be one of the libraries that brought about the concept of store management in application. Store management refers to the process of storing data provided by the  API coming from the backend to the frontend. Redux allows the data stored in 'store manager' to be available to all components of the application that need it.

### Key Takeaways
1. How to integrate redux with an application
2. Concept of store management
3. Meaning of action and reducer functions
4. How to connect the redux store to components.
5. Lastly, How to make asynchronous calls using redux thunk.

### References
https://redux.js.org/introduction/getting-started

### PR DESCRIPTION
**Pull request by Horlawhumy-dev**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x]  Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #3486.

### CODE REVIEW

**Comment by Horlawhumy-dev** [comment]
Yeah, I am done with the article.

**Comment by Horlawhumy-dev** [comment]
I just hope this article has not been left out @hectorkambow @ahmadmardeni1 @WanjaMIKE

**Comment by hectorkambow** [comment]
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

**Comment by Horlawhumy-dev** [comment]
Thanks, @M0nicah

**Comment by Horlawhumy-dev** [comment]
Yeah, I have done the plagiarism test. The updated article has been re-uploaded.

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

**Comment by hectorkambow** [comment]
@M0nicah  

I believe this PR have pass the plagiarism check and can go through prelim review.

**Comment by Horlawhumy-dev** [comment]
Yes sir. It has passed plagiarism test. With my best of knowledge, it only requires the reviewer to move it to the next stage.

File: content/articles/index.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+### Making Asynchronous Calls
+
+There are so many types of async calls that can be done in Redux, but today I want to reference an API call. Performing async operations like API calls must never be done in `reducers` as warned earlier.
+
+However, every asynchronous call involves a function that can take parameters from the component calling them and returns another function with `dispatch()` and `getState()?` arguments returned. 
+
+The data to be received from the API shall be dispatched as the payload together with the specific action type. Every dispatch methods call the Redux store which triggers the reducers based on the type of action received. Then the reducer returns the updated state required by the caller which gets to the component in form of properties i.e `Props`.
+
+To demonstrate an instance of asynchronous calls in the Redux application, I came up with the codes below.
```
Comment (M0nicah) [inline content/articles/index.md:229]:
```suggestion
To demonstrate an instance of asynchronous calls in the Redux application, I came up with the code below.
```

File: content/articles/index.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+In this article, I will be covering these sub-topics been the most essentials of the Redux itself. Join me to explore more on this journey.
+
+* What is Redux and Store Management?
+* How to set up the Redux Store.
+* Concepts of Action and Reducers in Redux Data Flow.
+* Connecting an application with Redux Store.
+* Finally, How to make asynchronous calls with Redux Thunk.
+
+### What is Redux and Store Management?
+Generally speaking, Redux is said to be a JavaScript library that depicts how the state of the application should be managed and accessed. Redux manages every application state, by providing a centralized store that needs to be accessed throughout the application. There are many libraries like flux, ember, etc but Redux data flow is clear and easy to integrate with. What is the application state?
```
Comment (M0nicah) [inline content/articles/index.md:17]: This paragraph seems a little bit long. Try to break the sentences to make them shorter.
Let each paragraph explain one concept for clear understanding.

**Review by M0nicah — CHANGES_REQUESTED** [review]
Hello @Horlawhumy-dev, I've gone through your PR  and there seems to be some problems. Please go through the comments and suggestions I have put forward so you could make your article better.

**Note:** The title of your article should be in lowercase.
Thanks.

**Comment by Horlawhumy-dev** [comment]
I have corrected the article and perfeced the hero issues. @M0nicah ma

**Comment by M0nicah** [comment]
great article @Horlawhumy-dev , please review the changes I made in the commits to help improve the flow and format for your future articles.

**Comment by Horlawhumy-dev** [comment]
Wow! I have gone through the changes. Thanks @M0nicah

**Pull request merged**